presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixMetadata.java

```diff
@@ -0,0 +1,388 @@
+import static org.apache.phoenix.util.SchemaUtil.getEscapedArgument;
+
+public class PhoenixMetadata
+        extends JdbcMetadata
+        implements ConnectorMetadata
+{
+    // col name used for PK if none provided in DDL
+    public static final String ROWKEY = "ROWKEY";
+    // Maps to Phoenix's default empty schema
+    public static final String DEFAULT_SCHEMA = "default";
+    private final PhoenixClient phoenixClient;
+
+    @Inject
+    public PhoenixMetadata(PhoenixClient phoenixClient)
+    {
+        super(phoenixClient, true);
+        this.phoenixClient = requireNonNull(phoenixClient, "client is null");
+    }
+
+    @Override
+    public List<String> listSchemaNames(ConnectorSession session)
+    {
+        return ImmutableList.copyOf(phoenixClient.getSchemaNames(JdbcIdentity.from(session)));
+    }
+
+    @Override
+    public JdbcTableHandle getTableHandle(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        return phoenixClient.getTableHandle(JdbcIdentity.from(session), schemaTableName).map(tableHandle ->
+                new JdbcTableHandle(
+                        schemaTableName,
+                        tableHandle.getCatalogName(),
+                        toPrestoSchemaName(Optional.ofNullable(tableHandle.getSchemaName())).orElse(null),
+                        tableHandle.getTableName()))
+                .orElse(null);
+    }
+
+    @Override
+    public List<ConnectorTableLayoutResult> getTableLayouts(ConnectorSession session, ConnectorTableHandle table, Constraint<ColumnHandle> constraint, Optional<Set<ColumnHandle>> desiredColumns)
+    {
+        JdbcTableHandle tableHandle = (JdbcTableHandle) table;
+        ConnectorTableLayout layout = new ConnectorTableLayout(new PhoenixTableLayoutHandle(tableHandle, constraint.getSummary(), desiredColumns));
+        return ImmutableList.of(new ConnectorTableLayoutResult(layout, constraint.getSummary()));
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata(ConnectorSession session, ConnectorTableHandle table)
+    {
+        return getTableMetadata(session, table, false);
+    }
+
+    public ConnectorTableMetadata getTableMetadata(ConnectorSession session, ConnectorTableHandle table, boolean rowkeyRequired)
+    {
+        JdbcTableHandle handle = (JdbcTableHandle) table;
+        ImmutableList.Builder<ColumnMetadata> columnMetadata = ImmutableList.builder();
+        List<JdbcColumnHandle> columns = rowkeyRequired ? phoenixClient.getColumns(session, handle) : phoenixClient.getNonRowkeyColumns(session, handle);
+        for (JdbcColumnHandle column : columns) {
+            columnMetadata.add(column.getColumnMetadata());
+        }
+        return new ConnectorTableMetadata(handle.getSchemaTableName(), columnMetadata.build(), getTableProperties(session, handle));
+    }
+
+    public Map<String, Object> getTableProperties(ConnectorSession session, JdbcTableHandle handle)
```

Comment:
rowkeys (with row_timestamp ) table properties isn't print from `show create table`.

Reply:
@combineads Added support for showing rowkeys in the table properties